client/src/app/api/api-details/api-details.component.ts

```diff
@@ -241,16 +241,19 @@ export class ApiDetailsComponent extends NavigableComponent implements OnDestroy
             this._runtimeVersion
           );
         })
-        .finally(() => {
-          this._disableSubmit = false;
-        })
-        .subscribe(() => {
-          this.clearBusy();
-          if (this.rrComponent) {
-            this.rrComponent.saveModel();
+        .subscribe(
+          () => {
+            this.clearBusy();
+            if (this.rrComponent) {
+              this.rrComponent.saveModel();
+            }
+            this.onReset();
+            this._disableSubmit = false;
+          },
+          () => {
+            this._disableSubmit = false;
           }
```

Comment:
This is not a suggestion to change what you did. I'm just curious how `subscribe` works:

```suggestion
          () => {
            this.clearBusy();
            if (this.rrComponent) {
              this.rrComponent.saveModel();
            }
            this.onReset();
          },
          () => {},
          () => {
            this._disableSubmit = false;
          }
```

Reply:
Hi Joe! Subscribe takes two parameters. First one is `.then` and second one `.catch`. I did not see 3rd option to accommodate `.finally`. I wanted to use ".finally" functionality but due to that reason, I am using both ".then" and ".catch" functionality here :)